crates/runtime/src/instance.rs

```diff
@@ -648,12 +902,18 @@ impl InstanceHandle {
             VMBuiltinFunctionsArray::initialized(),
         );
 
-        // Check initializer bounds before initializing anything.
-        check_table_init_bounds(instance)?;
-        check_memory_init_bounds(instance, data_initializers)?;
+        // Check initializer bounds before initializing anything. Only do this
+        // when bulk memory is disabled, since the bulk memory proposal changes
+        // instantiation such that the intermediate results of failed
+        // initializations are visible.
+        if !is_bulk_memory {
```

Comment:
Oh bummer :(

I thought that the bulk memory proposal just changed types of errors, not changing semantics like this. Passing around this `is_bulk_memory` boolean is sort of a bummer, but given this change in semantics I don't think we can do much about it.